hotrod/src/main/java/io/hyperfoil/hotrod/HotRodRunData.java

```diff
@@ -0,0 +1,82 @@
+package io.hyperfoil.hotrod;
+
+import java.time.Clock;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.function.Consumer;
+
+import io.hyperfoil.api.config.Benchmark;
+import io.hyperfoil.api.config.Scenario;
+import io.hyperfoil.api.session.Session;
+import io.hyperfoil.core.api.PluginRunData;
+import io.hyperfoil.core.impl.ConnectionStatsConsumer;
+import io.hyperfoil.hotrod.api.HotRodRemoteCachePool;
+import io.hyperfoil.hotrod.config.HotRod;
+import io.hyperfoil.hotrod.config.HotRodPluginConfig;
+import io.hyperfoil.hotrod.connection.HotRodRemoteCachePoolImpl;
+import io.netty.channel.EventLoop;
+import io.vertx.core.Future;
+
+public class HotRodRunData implements PluginRunData {
+
+   private final HotRodPluginConfig plugin;
+   private HotRodRemoteCachePool[] pool;
+
+   public HotRodRunData(Benchmark benchmark, EventLoop[] executors, int agentId) {
+      this.plugin = benchmark.plugin(HotRodPluginConfig.class);
+
+      List<String> allCaches = new ArrayList<>();
+      Map<String, List<String>> cacheNamesByUri = new HashMap<>();
+      for (HotRod hotRod : this.plugin.hotRod()) {
+         List<String> cacheNames = cacheNamesByUri.get(hotRod.getUri());
+         if (cacheNames == null) {
+            cacheNames = new ArrayList<>();
+         }
+         cacheNames.add(hotRod.getCacheName());
+         cacheNamesByUri.put(hotRod.getUri(), cacheNames);
+
+         // validation
+         if (allCaches.contains(hotRod.getCacheName())) {
+            throw new IllegalStateException(String.format("Duplicated cache: %s", hotRod.getCacheName()));
```

Comment:
TBH I can easily imagine a cross-site test where the cache names would be same...

Reply:
In this case, we should add the uri into the `hotrodRequest`. 
WDYT ?
```
name: hotrod-example
hotrod:
  uri: hotrod://localhost:11222
  cacheName: my-cache
usersPerSec: 1
duration: 5s
scenario:
  - example:
      - randomInt: cacheKey <- 1 .. 999
      - randomUUID: cacheValue
      - hotrodRequest:
          uri: hotrod://localhost:11222
          put: my-cache
```

Reply:
Partially fixed by the commit: HotRod with multiple uri